projects/ad411x_ad717x/common/ad411x_ad717x_asdz_qsys.tcl

```diff
@@ -0,0 +1,154 @@
+
+# spi_engine_execution
+
+add_instance spi_engine_execution_0 spi_engine_execution
+set_instance_parameter_value spi_engine_execution_0 {DATA_WIDTH} {32}
+set_instance_parameter_value spi_engine_execution_0 {NUM_OF_SDI} {1}
+set_instance_parameter_value spi_engine_execution_0 {SDI_DELAY} {0}
+
+# spi_engine_interconnect
+
+add_instance spi_engine_interconnect_0 spi_engine_interconnect
+set_instance_parameter_value spi_engine_interconnect_0 {DATA_WIDTH} {32}
+set_instance_parameter_value spi_engine_interconnect_0 {NUM_OF_SDI} {1}
+
+# bridges
+
+add_instance clock_bridge_0 altera_clock_bridge
+set_instance_parameter_value clock_bridge_0 {EXPLICIT_CLOCK_RATE} {0.0}
+set_instance_parameter_value clock_bridge_0 {NUM_CLOCK_OUTPUTS} {1}
+
+add_instance reset_bridge_0 altera_reset_bridge
+set_instance_parameter_value reset_bridge_0 {ACTIVE_LOW_RESET} {1}
+set_instance_parameter_value reset_bridge_0 {NUM_RESET_OUTPUTS} {1}
+set_instance_parameter_value reset_bridge_0 {SYNCHRONOUS_EDGES} {none}
+set_instance_parameter_value reset_bridge_0 {USE_RESET_REQUEST} {0}
+
+# spi_engine_offload
+
+add_instance spi_engine_offload_0 spi_engine_offload
+set_instance_parameter_value spi_engine_offload_0 {ASYNC_TRIG}    {1}
+set_instance_parameter_value spi_engine_offload_0 {ASYNC_SPI_CLK} {0}
+set_instance_parameter_value spi_engine_offload_0 {DATA_WIDTH}    {32}
+set_instance_parameter_value spi_engine_offload_0 {NUM_OF_SDI}    {1}
+
+# axi pwm gen
+
+add_instance trigger_generator axi_pwm_gen
+set_instance_parameter_value trigger_generator {N_PWMS} {1}
+set_instance_parameter_value trigger_generator {PULSE_0_PERIOD} {85}
+set_instance_parameter_value trigger_generator {PULSE_0_WIDTH} {1}
+
+
+# exported interface
+
+add_interface ad411x_spi_sclk       clock source
+add_interface ad411x_spi_cs         conduit end
+add_interface ad411x_spi_sdi        conduit end
+add_interface ad411x_spi_sdo        conduit end
+add_interface ad411x_spi_trigger    conduit end
+add_interface ad411x_spi_resetn     reset source
+
+set_interface_property ad411x_spi_cs      EXPORT_OF spi_engine_execution_0.if_cs
+set_interface_property ad411x_spi_sclk    EXPORT_OF spi_engine_execution_0.if_sclk
+set_interface_property ad411x_spi_sdi     EXPORT_OF spi_engine_execution_0.if_sdi
+set_interface_property ad411x_spi_sdo     EXPORT_OF spi_engine_execution_0.if_sdo
+set_interface_property ad411x_spi_trigger EXPORT_OF spi_engine_offload_0.if_trigger
+set_interface_property ad411x_spi_resetn  EXPORT_OF reset_bridge_0.out_reset
+
+add_connection axi_spi_engine_0.if_spi_resetn reset_bridge_0.in_reset
+
+# clocks
+
+add_interface ad411x_spi_clk        clock source
+set_interface_property ad411x_spi_clk     EXPORT_OF clock_bridge_0.out_clk
+
+add_connection sys_clk.clk axi_spi_engine_0.s_axi_clock
+add_connection sys_clk.clk axi_dmac_0.s_axi_clock
+add_connection sys_clk.clk trigger_generator.s_axi_clock
+
+add_connection sys_dma_clk.clk trigger_generator.if_ext_clk
```

Comment:
Maybe the clock source for the trigger generator should be the clock bridge that generates the spi_clk.